### DISCUSSION TITLE
problems installing and running zigbee2mqtt

### DISCUSSION THREAD

**Discussion by unevoljitelj** [Koenkk/zigbee2mqtt#19873] [General]
i am not an expert of any kind and i am unable to troubleshooth these new issues i didnt have before. my install of mint plus openhab crashed and i have to rebuild but cant run z2m

first problem is while installing when npm ci gives warning about npm version, where if 16 it will warn it needs 18, if 18 it will warn about 16 etc
then npm run build returns nothing
then i try npm start and it wants to start but returns
Error: network commissioning failed - panId collision detected (expected=6754, actual=24225)
wich whatever i use as pan id doesnt work
if i try docker version it will stop at panid also
i did manage to install zigbee2mqtt more then once previosly so i dont understand what is happening now
i run clean install of linux mint 21.2, while previous version was 20. i alos tried installing on ubuntu/arm and had same problems. what can i do to fix install? got 13 devices that i cant connect :/
i prefer running it as a service bcos i cant handle docker, barely made it run using instructions
thanks for any help

if running different distro will make this easier ill gladly do it

Zigbee2MQTT:error 2023-11-26 12:18:46: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2023-11-26 12:18:46: Failed to start zigbee
Zigbee2MQTT:error 2023-11-26 12:18:46: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2023-11-26 12:18:46: Exiting...
Zigbee2MQTT:error 2023-11-26 12:18:46: Error: network commissioning failed - panId collision detected (expected=43966, actual=14841)
    at ZnpAdapterManager.beginCommissioning (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/manager.ts:370:19)
    at ZnpAdapterManager.start (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/manager.ts:88:17)
    at Controller.start (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/controller/controller.ts:132:29)
    at Zigbee.start (/opt/zigbee2mqtt/lib/zigbee.ts:60:27)
    at Controller.start (/opt/zigbee2mqtt/lib/controller.ts:101:27)
    at start (/opt/zigbee2mqtt/index.js:107:5)

**Comment by oywino** [comment]
What purpose does npm serve in your equation?

**Reply by unevoljitelj** [reply]
i dont care about npm but its needed to install and run zigbee2mqtt? or maybe its nodejs, but something is making these two have an issue on newer distro.
since i posted this i tried installing it on multiple distros. it installs manualy or by script every time only on armbian bullseye, armbian jammy sometimes it does, but sometimes it doesnt. on mint 21 i just cant make it. its no problem using it on bullseye, but newer distro would be nice or something with a nice desktop :P.

**Comment by oywino** [comment]
> i dont care about npm but its needed to install and run zigbee2mqtt? 

Well, I run Z2M in a docker container and I never had to use any npm or nodejs.
